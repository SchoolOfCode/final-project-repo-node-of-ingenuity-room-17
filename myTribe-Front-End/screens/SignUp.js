import { StatusBar } from 'expo-status-bar';
import {
	StyleSheet,
	Text,
	View,
	ImageBackground,
	ScrollView,
	TextInput,
	Button,
	KeyboardAvoidingView,
	Dimensions,
} from 'react-native';
import headerImage from '../assets/header-background.jpeg';

const screenHeight = Dimensions.get('window').height;

export default function SignUp() {
	return (
		<KeyboardAvoidingView behavior='position' style={styles.container}>
			<ScrollView contentContainerStyle={styles.scrollView}>
				<ImageBackground
					resizeMode='cover'
					style={styles.header}
					source={headerImage}
				>
					<View style={styles.overlay}>
						<Text style={styles.heading}>Join our tribe.</Text>
						<Text style={styles.subHeading}>
							Start taking control of your household.
						</Text>
					</View>
				</ImageBackground>
				<View style={styles.registration}>
					<Text style={styles.sectionHeading}>Sign Up</Text>
					<Text style={styles.label}>Your name</Text>
					<TextInput style={styles.input} />
					<Text style={styles.label}>Email</Text>
					<TextInput style={styles.input} />
					<Text style={styles.label} secureTextEntry>
						Password
					</Text>
					<TextInput style={styles.input} />
					<Text style={styles.label}>Confirm password</Text>
					<TextInput style={styles.input} secureTextEntry />
					<View style={styles.btnContainer}>
						<Button
							buttonStyle={styles.btn}
							color='#FEB800'
							title='continue'
						/>
					</View>
				</View>
			</ScrollView>
		</KeyboardAvoidingView>
	);
}

const styles = StyleSheet.create({
	container: {
		height: '100%',
		width: '100%',
		justifyContent: 'space-around',
	},

	scrollView: {
		width: '100%',
		height: '100%',
		backgroundColor: '#F2F0F0',
	},

	header: {
		height: 250,
		width: '100%',
		resizeMode: 'cover',
	},

	overlay: {
		justifyContent: 'center',
		height: '100%',
		width: '100%',
		backgroundColor: 'rgba(0,0,0,0.3)',
		paddingLeft: 50,
		paddingRight: 50,
	},

	heading: {
		fontSize: 40,
		fontWeight: 'bold',
		color: 'white',
	},

	subHeading: {
		fontSize: 20,
		color: 'white',
	},

	registration: {
		width: '100%',
		height: screenHeight - 550,
		alignItems: 'flex-start',
		padding: 50,
		backgroundColor: '#F2F0F0',
	},

	sectionHeading: {
		fontSize: 24,
		fontWeight: 'bold',
		marginBottom: 50,
	},
	input: {
		width: '100%',
		height: 40,
		backgroundColor: 'white',
		borderRadius: 5,
		paddingLeft: 10,
		paddingRight: 10,
		marginBottom: 30,
	},
	label: {
		marginBottom: 10,
	},
	btnContainer: {
		fontWeight: 'bold',
		width: '100%',
		fontSize: 20,
		alignItems: 'flex-end',
		marginTop: 30,
	},
});
