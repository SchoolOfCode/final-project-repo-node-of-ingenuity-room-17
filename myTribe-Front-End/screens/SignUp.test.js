import React from "react";
import renderer from "react-test-renderer";
import { render, screen, FireEvent } from '@testing-library/react-native';
import {expect, it, jest} from '@jest/globals'
import SignUp from './SignUp'


it("check if SignUp screen contains a Sign Up button with Test ID signUpButton", async () => {
  const {getByTestId}= render(<SignUp navigation={{  getParam: jest.fn()  }} />);
  expect(getByTestId('signUpButton')).toBeDefined();
});

  it("check signup screen renders correctly", () => {
    const tree = renderer.create(<SignUp />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  it("check if SignUp screen contains text input element with test ID 'textInput", async () => {
 const {getByTestId}= render(<SignUp navigation={{  getParam: jest.fn()  }} />);

 expect(getByTestId('textInput')).toBeDefined();

});


