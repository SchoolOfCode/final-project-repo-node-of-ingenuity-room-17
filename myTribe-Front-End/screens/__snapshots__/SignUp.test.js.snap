// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUp /> check signup screen renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "height": "100%",
      "justifyContent": "space-around",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "bottom": 0,
      }
    }
  >
    <RCTScrollView
      contentContainerStyle={
        Object {
          "backgroundColor": "#F2F0F0",
          "height": "100%",
          "width": "100%",
        }
      }
    >
      <View>
        <View
          accessibilityIgnoresInvertColors={true}
          style={
            Object {
              "height": 250,
              "resizeMode": "cover",
              "width": "100%",
            }
          }
        >
          <Image
            resizeMode="cover"
            source={1}
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "height": 250,
                  "width": "100%",
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "rgba(0,0,0,0.3)",
                "height": "100%",
                "justifyContent": "center",
                "paddingLeft": 50,
                "paddingRight": 50,
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 40,
                  "fontWeight": "bold",
                }
              }
            >
              Join our tribe.
            </Text>
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 20,
                }
              }
            >
              Start taking control of your household.
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "backgroundColor": "#F2F0F0",
              "height": 784,
              "padding": 50,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
                "fontWeight": "bold",
                "marginBottom": 50,
              }
            }
          >
            Sign Up
          </Text>
          <Text
            style={
              Object {
                "marginBottom": 10,
              }
            }
          >
            Your name
          </Text>
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            rejectResponderTermination={true}
            style={
              Object {
                "backgroundColor": "white",
                "borderRadius": 5,
                "height": 40,
                "marginBottom": 30,
                "paddingLeft": 10,
                "paddingRight": 10,
                "width": "100%",
              }
            }
            testID="textInput"
            underlineColorAndroid="transparent"
          />
          <Text
            style={
              Object {
                "marginBottom": 10,
              }
            }
          >
            Email
          </Text>
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            rejectResponderTermination={true}
            style={
              Object {
                "backgroundColor": "white",
                "borderRadius": 5,
                "height": 40,
                "marginBottom": 30,
                "paddingLeft": 10,
                "paddingRight": 10,
                "width": "100%",
              }
            }
            underlineColorAndroid="transparent"
          />
          <Text
            secureTextEntry={true}
            style={
              Object {
                "marginBottom": 10,
              }
            }
          >
            Password
          </Text>
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            rejectResponderTermination={true}
            secureTextEntry={true}
            style={
              Object {
                "backgroundColor": "white",
                "borderRadius": 5,
                "height": 40,
                "marginBottom": 30,
                "paddingLeft": 10,
                "paddingRight": 10,
                "width": "100%",
              }
            }
            underlineColorAndroid="transparent"
          />
          <Text
            style={
              Object {
                "marginBottom": 10,
              }
            }
          >
            Confirm password
          </Text>
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            rejectResponderTermination={true}
            secureTextEntry={true}
            style={
              Object {
                "backgroundColor": "white",
                "borderRadius": 5,
                "height": 40,
                "marginBottom": 30,
                "paddingLeft": 10,
                "paddingRight": 10,
                "width": "100%",
              }
            }
            underlineColorAndroid="transparent"
          />
          <Text
            style={
              Object {
                "color": "black",
                "marginBottom": 10,
                "textAlign": "center",
              }
            }
          >
            
          </Text>
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "fontSize": 20,
                "fontWeight": "bold",
                "marginTop": 30,
                "width": "100%",
              }
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={Object {}}
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {},
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#007AFF",
                        "fontSize": 18,
                        "margin": 8,
                        "textAlign": "center",
                      },
                      Object {
                        "color": "#FEB800",
                      },
                    ]
                  }
                >
                  continue
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;

exports[`check signup screen renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "height": "100%",
      "justifyContent": "space-around",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "bottom": 0,
      }
    }
  >
    <RCTScrollView
      contentContainerStyle={
        Object {
          "backgroundColor": "#F2F0F0",
          "height": "100%",
          "width": "100%",
        }
      }
    >
      <View>
        <View
          accessibilityIgnoresInvertColors={true}
          style={
            Object {
              "height": 250,
              "resizeMode": "cover",
              "width": "100%",
            }
          }
        >
          <Image
            resizeMode="cover"
            source={1}
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "height": 250,
                  "width": "100%",
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "rgba(0,0,0,0.3)",
                "height": "100%",
                "justifyContent": "center",
                "paddingLeft": 50,
                "paddingRight": 50,
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 40,
                  "fontWeight": "bold",
                }
              }
            >
              Join our tribe.
            </Text>
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 20,
                }
              }
            >
              Start taking control of your household.
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "backgroundColor": "#F2F0F0",
              "height": 784,
              "padding": 50,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
                "fontWeight": "bold",
                "marginBottom": 50,
              }
            }
          >
            Sign Up
          </Text>
          <Text
            style={
              Object {
                "marginBottom": 10,
              }
            }
          >
            Your name
          </Text>
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            rejectResponderTermination={true}
            style={
              Object {
                "backgroundColor": "white",
                "borderRadius": 5,
                "height": 40,
                "marginBottom": 30,
                "paddingLeft": 10,
                "paddingRight": 10,
                "width": "100%",
              }
            }
            testID="textInput"
            underlineColorAndroid="transparent"
          />
          <Text
            style={
              Object {
                "marginBottom": 10,
              }
            }
          >
            Email
          </Text>
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            rejectResponderTermination={true}
            style={
              Object {
                "backgroundColor": "white",
                "borderRadius": 5,
                "height": 40,
                "marginBottom": 30,
                "paddingLeft": 10,
                "paddingRight": 10,
                "width": "100%",
              }
            }
            underlineColorAndroid="transparent"
          />
          <Text
            secureTextEntry={true}
            style={
              Object {
                "marginBottom": 10,
              }
            }
          >
            Password
          </Text>
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            rejectResponderTermination={true}
            secureTextEntry={true}
            style={
              Object {
                "backgroundColor": "white",
                "borderRadius": 5,
                "height": 40,
                "marginBottom": 30,
                "paddingLeft": 10,
                "paddingRight": 10,
                "width": "100%",
              }
            }
            underlineColorAndroid="transparent"
          />
          <Text
            style={
              Object {
                "marginBottom": 10,
              }
            }
          >
            Confirm password
          </Text>
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            rejectResponderTermination={true}
            secureTextEntry={true}
            style={
              Object {
                "backgroundColor": "white",
                "borderRadius": 5,
                "height": 40,
                "marginBottom": 30,
                "paddingLeft": 10,
                "paddingRight": 10,
                "width": "100%",
              }
            }
            underlineColorAndroid="transparent"
          />
          <Text
            style={
              Object {
                "color": "black",
                "marginBottom": 10,
                "textAlign": "center",
              }
            }
          >
            
          </Text>
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "fontSize": 20,
                "fontWeight": "bold",
                "marginTop": 30,
                "width": "100%",
              }
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={Object {}}
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="signUpButton"
            >
              <View
                style={
                  Array [
                    Object {},
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#007AFF",
                        "fontSize": 18,
                        "margin": 8,
                        "textAlign": "center",
                      },
                      Object {
                        "color": "#FEB800",
                      },
                    ]
                  }
                >
                  continue
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
